src/js/trove/matrix.js

```diff
@@ -0,0 +1,1310 @@
+        //subtract matrix
+        var subMatrix = function (self, other) {
+            runtime.ffi.checkArity(2, arguments, "sub-mat", false);
+            runtime.checkArgsInternal2("matrix", "sub-mat", self, annMatrix, other, annMatrix);
+            if (!sameDims(self, other)) {
+                return runtime.ffi.throwMessageException("Matrices have dimensions " + printDims(self) + " and " + printDims(other) + " . They cannot be added");
+            } else {
+                new_arr = new Array(self.$l);
+                for (var i = 0; i < self.$l; i++) {
+                    new_arr[i] = runtime.minus(self.$underlyingMat[i], other.$underlyingMat[i]);
+                }
+
+                return createMatrixFromArray(self.$h, self.$w, new_arr);
+            }
+        }
+
+        //multiply matrix
+        var multMatrix = function (self, other) {
+            runtime.ffi.checkArity(2, arguments, "mult-mat", false);
+            runtime.checkArgsInternal2("Matrix", "mult-mat", self, annMatrix, other, annMatrix);
+            if (self.$w != other.$h) {
+                return runtime.ffi.throwMessageException("The width of the first matrix and the height of the second matrix need to be equal");
+
+            } else {
+                new_arr = new Array((self.$h * other.$w));
+                for (var i = 0; i < self.$h; i++) {
+                    for (var j = 0; j < other.$w; j++) {
+                        var elm = runtime.makeNumber(0);
+                        for (var k = 0; k < self.$w; k++) {
+                            elm = runtime.plus(elm, runtime.times(get1dElem(self, i, k), get1dElem(other, k, j)));
+                        }
+                        new_arr[get1dpos(i, j, other.$w)] = elm;
+                    }
+                }
+                return createMatrixFromArray(self.$h, other.$w, new_arr);
+            }
+        }
+        //Output given matrix
+        var outputMatrix = runtime.makeMethod0(function (self) {
+            arity(1, arguments, "_output", false);
+            runtime.checkArgsInternalInline("Matrix", "_output", self, annMatrix);
+            var rows = [];
+            var matr = self.$underlyingMat;
+            var vsValue = get(VS, "vs-value");
+            for (var i = 0; i < matr.length; i++) {
+                rows.push(vsValue.app(matr[i]));
+
+            }
+            return get(VS, "vs-collection").app(
+                runtime.makeString("mat" + printDims(self)),
+                runtime.ffi.makeList(rows))
+        });
+
+        //Get matrix dimensions as a tuple 
+        var getMatrixDims = runtime.makeFunction(function (self) {
+            runtime.ffi.checkArity(1, arguments, "dims-mat", false);
+            runtime.checkArgsInternal1("Matrix", "dims-mat", self, annMatrix);
+            return runtime.makeTuple([self.$h, self.$w]);
+        }, "dims-mat");
+
+        //get element at (h,c)
+        var getMatrixElms = runtime.makeFunction(function (self, h, c) {
```

Comment:
What do `h` and `c` mean?  I can imagine row and column, x and y, width and height...but height and column don't seem to go together...